unsafe/src/main/java/org/apache/spark/unsafe/bitset/BitSetMethods.java

```diff
@@ -71,8 +72,11 @@ public static boolean isSet(Object baseObject, long baseOffset, int index) {
    * Returns {@code true} if any bit is set.
    */
   public static boolean anySet(Object baseObject, long baseOffset, long bitSetWidthInBytes) {
-    for (int i = 0; i <= bitSetWidthInBytes; i++) {
-      if (PlatformDependent.UNSAFE.getByte(baseObject, baseOffset + i) != 0) {
+    assert bitSetWidthInBytes % SIZE_OF_LONG == 0;
+    int widthInLong = (int)(bitSetWidthInBytes / SIZE_OF_LONG);
```

Comment:
SIZE_OF_LONG is measured in bits, not bytes, so I don't think that this is correct either.

Given the number of minor issues like this, can you please add a proper unit test for this method?